lib/logstash/plugin_mixins/jdbc/jdbc.rb

```diff
@@ -195,14 +198,22 @@ def open_jdbc_connection
       end
     end
 
+    public
+    def prepare_jdbc_connection
+      @connection_lock = ReentrantLock.new
```

Comment:
Is there a reason to use `ReentrantLock` instead of `Mutex`? I think with `Mutex` we could use its `synchronize` block and not have to explicitly handle unlocking with `ensure`.

Reply:
This comes from the reverted commit, and is just returning what was there before, but I agree with you. The `ReentrantLock#synchronize` would still be useful for clarity. I'd be glad to refactor.

Reply:
oh, wow. Yeah that is absolutely an issue. Nice analysis.